lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -45,24 +45,28 @@ def __init__(self, config: PbenchServerConfig, logger: logging.Logger):
             role=API_OPERATION.READ,
         )
 
-    def _get(self, json_data: JSON, request: Request) -> Response:
+    def _get(self, _, request: Request) -> Response:
         """
         Get a list of datasets matching a set of criteria.
 
         NOTE: This does not rely on a JSON payload to identify the dataset and
         desired metadata keys; instead we rely on URI query parameters.
 
+        Args:
+            request: The original Request object containing query parameters
+
         GET /api/v1/datasets/list?start=1970-01-01&end=2040-12-31&owner=fred&metadata=dashboard.seen,server.deletion
         """
 
         # We missed automatic schema validation due to the lack of a JSON body;
         # construct an equivalent JSON body now so we can run it through the
         # validator.
-        new_json = self._validate_query_params(request, self.schema)
+        json = self._collect_query_params(request, self.schema)
+        new_json = self.schema.validate(json) if json else json
```

Comment:
This is sort of an obfuscation (or else it's much more subtle than it might otherwise be...).  I think it would be clearer if it were styled like:
```
        new_json = self.schema.validate(json) if json else {}
```
That is, if `json` is empty, lets set `new_json` to something which is obvious-to-the-reader empty, instead of this already empty thing which we happen to have lying around.

Ditto in `lib/pbench/server/api/resources/datasets_metadata.py` and `lib/pbench/server/api/resources/datasets_daterange.py`.

Reply:
If !json, then json already is {}... but `validate` will fail with an empty payload and I didn't want to change that. I suppose if I have to change this I'll do what I avoided doing initially:

```
    new_json = json
    if json:
        new_json = self.schema.validate(json)
```

... which avoids all ambiguity at the expense of a few extra pixels.

Reply:
> If `!json`, then json already is `{}`

Agreed....

> ... but `validate` will fail with an empty payload

...agreed...but `validate()` won't be called, so that's not a problem, right?

I don't think your proposed revision is better than using the ternary.  (And, no, you don't _have_ to change this!)

If we call `validate()`, it will return a new dictionary; so, all I was saying is that it would be better to assign `new_json` to a fresh empty dictionary than to bind it to the value that `json` has, because this is clearer.